### PR TITLE
Start query

### PR DESCRIPTION
**Pull request by nicpottier**
Adds the nullable DB field `query` to `flows_flowstart` which mailroom will honor to start flows based on a query.

### CODE REVIEW

File: temba/flows/models.py [hunk 1]
```diff
@@ -4744,6 +4744,9 @@ class FlowStart(models.Model):
     # the individual contacts that should be considered for start in this flow
     contacts = models.ManyToManyField(Contact)
 
+    # the query (if any) that should be used to select contacts to start
+    query = models.CharField(max_length=255, null=True)
```
Comment (rowanseymour) [inline temba/flows/models.py]: note that `ContactGroup.query` doesn't have a length limit (it's a TextField) and we've definitely seen some crazy searches.. but maybe it should have a limit?

Comment (nicpottier) [reply temba/flows/models.py]: Django requires CharField to have a limit but happy to make this a TextField, as you said users get crazy, especially in flows.

Comment (rowanseymour) [reply temba/flows/models.py]: Yeah I'd lean toward a TextField and then maybe define MAX_QUERY_LEN somewhere to check for both dynamic groups and flow starts.

Comment (nicpottier) [reply temba/flows/models.py]: Cool, changed it to TextField. Punting on max query length since that hasn't been a problem thus far.

**Pull request merged**